### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by NegativeScript**

### CODE REVIEW

File: furusiya/function_mess.py [hunk 1]
```diff
@@ -223,18 +222,13 @@ def play_game():
...
-    Game.ui.con.clear()  # unexplored areas start black (which is the default background color)
 
     while not tdl.event.is_window_closed():
-
         # draw all objects in the list
         Game.renderer.render()
 
-        # erase all objects at their old locations, before they move
-        for obj in Game.area_map.entities:
-            obj.clear()
```
Comment (nightblade9) [inline furusiya/function_mess.py]: Why isn't this necessary any more?

File: furusiya/main.py [hunk 2]
```diff
@@ -37,16 +39,16 @@ def main_menu():
         if choice == 1:  # load last game
             try:
                 load_game()
-            except:
+            except Exception as e:
```
Comment (nightblade9) [inline furusiya/main.py]: I don't see any value in addin this if we're not doing anything with `e`. Can't we collapse these two `except` clauses together?

File: furusiya/main.py [hunk 3]
```diff
@@ -37,16 +39,16 @@ def main_menu():
...
         elif choice == 2:  # quit
             break
 
-if config.has("seed"):
-    seed = config.get("seed")
-    random.seed(seed)
-    print("Seeding as universe #{}".format(seed))
-    
+
+seed = config.get("seed") or int(datetime.now().timestamp())
```
Comment (nightblade9) [inline furusiya/main.py]: This is not necessary. I verified it on the REPL.

1) `random.seed` does not need to be called; if it isn't, it's seeded with a random seed.
2) If it's called with `None`, it gets seeded with a random seed

I would suggest removing this change and instead adding a comment: if there is no seed, this gets a random seed.

File: furusiya/model/factories/item_factory.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 from model.item import Item
 
 
-def create_item(x, y, character, name, color, use_function):
+def create_item(x, y, character, name, color, use_function=None):
```
Comment (nightblade9) [inline furusiya/model/factories/item_factory.py]: Good catch!

File: furusiya/model/maps/area_map.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ def is_walkable(self, x, y):
                         ]) == 0)
 
     def is_visible_tile(self, x, y):
-        return self.is_walkable(x, y) and not self.tiles[x][y].block_sight
+        return self.is_on_map(x, y) and not self.tiles[x][y].block_sight
```
Comment (nightblade9) [inline furusiya/model/maps/area_map.py]: I don't agree with this. It means everything is visible unless it explicitly blocks sight. Things that are not walkable should not be visible. Please revert it, and set monsters to `block_sight = False` instead.

File: furusiya/function_mess.py (hunk 1 above)
Comment (NegativeScript) [reply furusiya/function_mess.py]: See map_renderer.py:
https://github.com/nightblade9/furusiya/blob/342ec2e51eedbd70fcd58672b04d1ea6729b39da/furusiya/view/map_renderer.py#L53-L61

Old monster locations are overridden if they're in the FOV. If they're no longer in it:
https://github.com/nightblade9/furusiya/blob/342ec2e51eedbd70fcd58672b04d1ea6729b39da/furusiya/view/map_renderer.py#L30-L36
They are rendered as "explored".

So yes, unneeded.

File: furusiya/main.py (hunk 2 above)
Comment (NegativeScript) [reply furusiya/main.py]: There's only one `except` clause.
Admittedly, I added `Exception as e` when I was debugging, and forgot to remove it. It's doing no harm, though.

File: furusiya/main.py (hunk 3 above)
Comment (NegativeScript) [reply furusiya/main.py]: Yes, it does get a random seed, but it doesn't print it. This does.

File: furusiya/model/maps/area_map.py (hunk 5 above)
Comment (NegativeScript) [reply furusiya/model/maps/area_map.py]: Let me just post a line of code:
https://github.com/nightblade9/furusiya/blob/342ec2e51eedbd70fcd58672b04d1ea6729b39da/furusiya/model/maps/area_map.py#L71
A tile blocks sight in two cases:
- It is explicitly made to do so.
- It isn't walkable and hasn't been explicitly made not sight-blocking.

So by default, non-walkable tiles block sight *unless* explicitly allowed. Sounds better?

File: furusiya/main.py (hunk 2 above)
Comment (nightblade9) [reply furusiya/main.py]: My bad, I read this wrong. This is fine with me. 👍

File: furusiya/main.py (hunk 3 above)
Comment (nightblade9) [reply furusiya/main.py]: Yes, but your seed may not be as good as the built-in one. Unfortunately, it seems like there's just no way to get the built-in seed.

How about `random.seed(random.random())` instead?

File: furusiya/model/maps/area_map.py (hunk 5 above)
Comment (nightblade9) [reply furusiya/model/maps/area_map.py]: Yes, sorry. This is fine.

**Pull request merged**